store/postgres/src/chain_store.rs

```diff
@@ -2308,15 +2309,25 @@ impl ChainStoreTrait for ChainStore {
     }
 
     fn set_chain_identifier(&self, ident: &ChainIdentifier) -> Result<(), Error> {
+        use primary::chains as c;
         use public::ethereum_networks as n;
 
         let mut conn = self.pool.get()?;
-        diesel::update(n::table.filter(n::name.eq(&self.chain)))
-            .set((
-                n::genesis_block_hash.eq(ident.genesis_block_hash.hash_hex()),
-                n::net_version.eq(&ident.net_version),
-            ))
-            .execute(&mut conn)?;
+        conn.transaction(|conn| {
+            diesel::update(n::table.filter(n::name.eq(&self.chain)))
+                .set((
+                    n::genesis_block_hash.eq(ident.genesis_block_hash.hash_hex()),
+                    n::net_version.eq(&ident.net_version),
+                ))
+                .execute(conn)?;
+
+            diesel::update(c::table.filter(c::name.eq(&self.chain)))
+                .set((
+                    c::genesis_block_hash.eq(ident.genesis_block_hash.hash_hex()),
+                    c::net_version.eq(&ident.net_version),
+                ))
+                .execute(conn)
+        })?;
```

Comment:
Bit confused about the location of these tables. Isn't the `chains` table on the `primary` shard and `ethereum_network` on the shard of the block cache for the chain is, so in this transaction we are only updating the block cache shard right?
Don't we need to update in primary also?